pkg/utils/kubernetes/client/client.go

```diff
@@ -201,6 +206,57 @@ func (cl *cleaner) doClean(ctx context.Context, c client.Client, obj client.Obje
 	return nil
 }
 
+var _ Cleaner = (*volumeSnapshotContentCleaner)(nil)
+
+type volumeSnapshotContentCleaner struct {
+	defaultCleaner Cleaner
+}
+
+// Clean annotates, deletes and optionally finalizes the VolumeSnapshotContents that expired their termination date.
+func (v *volumeSnapshotContentCleaner) Clean(ctx context.Context, c client.Client, obj runtime.Object, opts ...CleanOption) error {
+	cleanOptions := &CleanOptions{}
+	cleanOptions.ApplyOptions(opts)
+
+	switch o := obj.(type) {
+	case client.ObjectList:
+		err := cleanCollectionAction(ctx, c, o, cleanOptions, v.addDeleteAnnotation)
+		if err != nil {
+			return err
+		}
+	case client.Object:
+		err := cleanAction(ctx, c, o, cleanOptions, v.addDeleteAnnotation)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("type %T does neither implement client.Object nor client.ObjectList", obj)
+	}
+
+	return defaultCleaner.Clean(ctx, c, obj, opts...)
+}
+
+const annVolumeSnapshotBeingDeleted = "snapshot.storage.kubernetes.io/volumesnapshot-being-deleted"
+
+func (v volumeSnapshotContentCleaner) addDeleteAnnotation(ctx context.Context, c client.Client, obj client.Object, _ *CleanOptions) error {
```

Comment:
You may also want to remove the annotation `snapshot.storage.kubernetes.io/volumesnapshot-being-created` if it is set. See [comment in issue 1505](https://github.tools.sap/kubernetes-canary/issues-canary/issues/1505#issuecomment-471097) where setting the `volumesnapshot-being-deleted` was not sufficient.